index.js

```diff
@@ -58,6 +59,9 @@ async function run() {
       balances: {}, // stores account balances like this { [colorIndex]: { address1: 0, ... } }
       unspent: {}, // stores unspent outputs (deposits, transfers)
       processedDeposit: 0,
+      epoch: 0,
+      lastEpochHeight: 0,
+      epochLength: null,
```

Comment:
`0` is treated as `null` in comparisons, why not use it?

Reply:
other idea: bridge contract and node can not work without at least 1 validator. couldn't initial epochLength be 1?

Reply:
Also, we can read events before setting up `initialState` and put value from the first event

Reply:
how do you read the first event that happened in a contract? wouldn't you need to know the block-height when it was deployed?